src/ripple/test/impl/JSONRPCClient.cpp

```diff
@@ -116,6 +118,12 @@ class JSONRPCClient : public AbstractClient
         {
             Json::Value jr;
             jr["method"] = cmd;
+            if (rpc_version_ == 2)
+            {
+                jr["jsonrpc"] = "2.0";
+                jr["ripplerpc"] = "2.0";
+                jr["id"] = 5;
```

Comment:
I have no problem with an atomic counter. I don't think there's significant overhead with an atomic on a single thread, which I'm pretty sure _all_ of our current jtx tests are.

> And I fear we may have already approached that boundary.

I disagree. The test framework is highly abstracted, but it's also very robust so far, as evidenced by how few false negatives we have running the jtx unittests.

> We are currently checking that the correct id exists in the individual tests

Which, IMHO, leads to a lot of unnecessarily duplicated code, and with all requests using the same ID is also IMHO not very valuable (e.g. a response to the wrong request wouldn't be detected this way).